### PR TITLE
radosgw-admin: add --uid check in bucket list command

### PR DESCRIPTION
**Pull request by vumrao**
radosgw-admin: add --uid check in bucket list command
Fixes: https://tracker.ceph.com/issues/41589

### CODE REVIEW

**Comment by vumrao** [comment]
- The vstart cluster results:

~~~
[root@3cfea3e7c7e2 build]# bin/radosgw-admin user list
[
    "56789abcdef0123456789abcdef0123456789abcdef0123456789abcdef01234",
    "testx$9876543210abcdef0123456789abcdef0123456789abcdef0123456789abcdef",
    "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef",
    "test",
    "testid"
]
~~~

- Not providing value to the argument --uid

~~~
[root@3cfea3e7c7e2 build]# bin/radosgw-admin bucket list --uid=
bad uid
~~~

- Providing a user which does not exist

~~~
[root@3cfea3e7c7e2 build]# bin/radosgw-admin bucket list --uid=xyz
ERROR: could not find user: xyz
~~~

- User with buckets

~~~
[root@3cfea3e7c7e2 build]# bin/radosgw-admin bucket list --uid=testid
[
    "my-new-bucket1",
    "my-new-bucket2",
    "my-new-bucket3",
    "my-new-bucket4",
    "my-new-bucket5"
]
~~~

- User without buckets

~~~
[root@3cfea3e7c7e2 build]# bin/radosgw-admin bucket list --uid=test
[]
~~~

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -2919,6 +2919,10 @@ int main(int argc, const char **argv)
       break;
     } else if (ceph_argparse_witharg(args, i, &val, "-i", "--uid", (char*)NULL)) {
       user_id.from_str(val);
+      if (user_id.empty()) {
```
Comment (mattbenjamin) [inline src/rgw/rgw_admin.cc]: maybe "no value for uid" given the test

Comment (vumrao) [reply src/rgw/rgw_admin.cc]: Thanks, Matt. Done.

**Comment by vumrao** [comment]
- After Matt suggestion test result from vstart cluster:

~~~
# bin/radosgw-admin bucket list --uid=
no value for uid
~~~

**Comment by cbodley** [comment]
passed qa in http://pulpito.ceph.com/cbodley-2019-09-18_14:51:41-rgw-wip-cbodley-testing-distro-basic-smithi/

(slow ops failures tracked in https://tracker.ceph.com/issues/41834)

**Comment by vumrao** [comment]
Thank you @cbodley.

**Pull request merged**